### PR TITLE
Add `cupy.setdiff1d` api.

### PR DESCRIPTION
**Pull request by khushi-411**
Hi,
The PR adds `cupy.setdiff1d1`. It follows #6078. Always a pleasure to hear from you all.
Thanks!

### CODE REVIEW

File: tests/cupy_tests/logic_tests/test_truth.py [hunk 1]
```diff
@@ -125,6 +125,30 @@ def test(self, xp, dtype):
...
+        y = xp.array([8, 7, 3, 9, 0], dtype=dtype)
+        return xp.setdiff1d(x, y)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_setdiff1d_3(self, xp, dtype):
+        x = xp.array([3, 4, 9, 1, 5, 4], dtype=dtype)
+        y = xp.array([8, 7, 3, 9, 0], dtype=dtype)
+        return xp.setdiff1d(x, y, assume_unique=True)
+
```
Comment (emcastillo) [inline tests/cupy_tests/logic_tests/test_truth.py:150]: Can you add more exhaustive test cases?
if the routine deals with arrays with duplicate elements we expect to see that tested as well.
What happens if you feed a 2d array? or a simple scalar? are the errors consistent with NumPy ones?
you should test not only 1 or 2 working cases but cover all the edge cases and erratic behavior.

Comment (khushi-411) [reply tests/cupy_tests/logic_tests/test_truth.py:150]: Oops, good points. I'll update soon. Thanks!

**Comment by asi1024** [comment]
Could you resolve conflicts?

**Comment by khushi-411** [comment]
Resolved! Thanks, @asi1024!

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
@khushi-411 LGTM! Thanks!

**Comment by khushi-411** [comment]
Hi, @asi1024! Hi, @emcastillo!
Thanks for reviewing! :)
